Add info about starting a workspace from raw devfile location (#2563)
codenvy/che-docs: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc

. Optional: Visit your {prod-short} dashboard pages to authenticate to your organization's instance of {prod-short}.

. Visit the URL to start a new workspace using the basic syntax:

> .. Using a Git repository URL:

Reviewer: ```suggestion
.. Use a Git repository URL:
```

+
The workspace has a unique URL: `pass:c,a,q[{prod-url}]#workspace__<unique_url>__`.

.. Using a direct URL to `devfile`:
[source,subs="+quotes,+attributes,+macros"]
+
----
pass:c,a,q[{prod-url}]#__<url_to_devfile>__
----
+
[TIP]
====

> You can pass Personal Access Token to the URL to have access to private repositories:

Reviewer 2: maybe we can add info that this would work for GitHub, GitLab, Bitbucket, Azure, and other types repositories that support this type of OAuth

Author: done

+
The workspace has a unique URL: `pass:c,a,q[{prod-url}]#workspace__<unique_url>__`.

> .. Using a direct URL to `devfile`:

Reviewer: ```suggestion
.. Use a direct URL to `devfile`:
```

> .. Using a direct URL to `devfile`:

Reviewer 3: FYI, this step is now in the wrong place: it's currently placed after the workspace has been started.

I also think this step makes the procedure difficult to follow and less readable.

In my personal opinion, the best solution to this update is to turn this page file into an assembly with two procedures in two separate `proc_...` files as follows:
1. Start a new branch from `main` (not from the branch of this PR).
2. In the new assembly file, keep the original heading and original file name (`starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc`), the original three intro paragraphs, and move up to them both the original bottom tip about browser bookmark managers and original *Additional resources*.
3. You can think of some parallel headings for two new procedure files, e.g.:
```
= Starting a new workspace from a Git-repository URL
= Starting a new workspace from a devfile URL
```
4. Under *Additional resources* in the assembly file, add two include statements for the two `proc_...` files.
5. Introduce the two new procedures in the intro of the assembly file as unordered list items.
6. Move the original content starting from the first tip (*"You can also use the Git Repo URL ..."*) up to and including *"The workspace has a unique URL"* to a new `proc_...` file that has the heading like `= Starting a new workspace from a Git-repository URL`.
7. There, move and tweak the following sentence into a new *Verification* section as a separate verification step (will be the same for both procedures):
```
A clone of the Git repository is present in the filesystem of the new workspace.
```
8. There, consider turning the following sentence into a NOTE (will be the same for both procedures):
```
The workspace has a unique URL: https://<che_fqdn>#workspace<unique_url>.
```
9. After you finalize the first procedure file, copy all its content into the second procedure file, and there incorporate the new content from this PR that's currently in step 2b) *"Using a direct URL to devfile:"* and as related to it elsewhere.

Reviewer: thanks for the input, I was thinking along similar lines but didn't have as precise a picture in my head as you just described.

[source,subs="+quotes,+attributes,+macros"]
+
----
pass:c,a,q[{prod-url}]#__<url_to_devfile>__
----
+
[TIP]
====
You can pass Personal Access Token to the URL to have access to private repositories:
[source,subs="+quotes,+attributes,+macros"]
----

> pass:c,a,q[{prod-url}]#__https://__<token>__@__<host>__/__<path> <1>

Reviewer 2: @vinokurig probably here we need to add `IMPORTANT:` note that devfile should contain the project reference, also maybe worth mentioning that path = path to devfile

----
<1> See xref:using-a-git-provider-access-token.adoc[].

> This would work for GitHub, GitLab, Bitbucket, Microsoft Azure, and other providers that support Personal Access Token.

Reviewer: ```suggestion
This works for GitHub, GitLab, Bitbucket, Microsoft Azure, and other providers that support Personal Access Token.
```

====
+

> .A URL for starting a new workspace from devfile location

Reviewer: This section gets rendered as an example. Is it the correct formatting? 
![image](https://user-images.githubusercontent.com/80100533/221644435-8ca8b2fb-30f6-400c-a7fe-5cc1514a3f89.png)

Author: Yes, this is an example with a Github link

Reviewer: understood, thanks!
